lib/elixir/lib/inspect/algebra.ex

```diff
@@ -396,7 +396,7 @@ defmodule Inspect.Algebra do
   def to_doc_with_opts(term, opts)
 
   def to_doc_with_opts(%_{} = struct, %Inspect.Opts{inspect_fun: fun} = opts) do
-    if opts.structs do
+    if opts.structs and Inspect.Map.valid_struct?(struct) do
```

Comment:
Should we move the function to this module instead?

Reply:
Indeed makes sense, I initially thought I'd need it at several places but not anymore 👍

Reply:
[59d3443](https://github.com/elixir-lang/elixir/pull/14718/commits/59d34430bdd042584c67358404a0c97f5c34e01c)